game-core/src/main/java/games/strategy/triplea/ui/TerritoryDetailPanel.java

```diff
@@ -108,6 +112,14 @@ public void setGameData(final GameData data) {
     territoryChanged(null);
   }
 
+  public void addAdditionalTerritoryDetailsFunction(final Function<Territory, String> method) {
```

Comment:
These two methods (`addAdditionalTerritoryDetailsFunction` and `removeAdditionalTerritoryDetailsFunction`) allow the ai to add text to the territory details panel.  This, with the heat map overlay, makes it really easy to see what the territory valuations are.